docs/sources/tempo/configuration/_index.md

```diff
@@ -707,6 +710,34 @@ query_frontend:
 
 ```
 
+### Limit query size to improve performance and stability
+
+Querying large tracing data presents several challenges.
+Span sets with large number of spans impact query performance and stability.
+In a similar manner, excessive queries result size can also negatively impact query performance.
+
+#### Limit the spans per spanset
+
+You can set the maximum spans per spanset by setting `max_spans_per_span_set` for the query-frontend.
+The default value is 100. Tempo and Grafana Enterprise Traces users can set this option in their configuration file.
+
+In Grafana or Grafana Cloud, you can set an equivalent option by using the **Span Limit** field in the [TraceQL query editor](https://grafana.com/docs/grafana-cloud/connect-externally-hosted/data-sources/tempo/query-editor/) in Grafana Explore.
+This field sets the maximum number of spans to return for each span set.
+If the **Span Limit** is set to `0`, then the capability is disabled.
```

Comment:
I'm not sure if this line is correct. From the change I've seen that if the configuration `max_spans_per_span_set` is set to `0`, then the maximum is disabled.

It'd be nice for the team to confirm.

```suggestion
To disable the maximum spans per span set limit, set `max_spans_per_span_set` to `0`.
```
Something like this, if correct.

Reply:
Okay. The text I wrote was based on [this comment thread](https://github.com/grafana/tempo/pull/4383#discussion_r1858731357) in the original PR, where Joe notes that you can adjust Span Limit "to whatever works for your or (or turned off with a 0)."

Reply:
```suggestion
To disable the maximum spans per span set limit, set `max_spans_per_span_set` to `0`.
When set to `0`, there is no maximum and users can put any value in **Span Limit**. 
However, this can only be set by a Tempo administrator, not by the user. 
```

Reply:
Kim, your last suggested change is correct 👍